provider/pkg/await/deployment.go

```diff
@@ -372,16 +395,16 @@ func (dia *deploymentInitAwaiter) processDeploymentEvent(event watch.Event) {
 
 	deployment, isUnstructured := event.Object.(*unstructured.Unstructured)
 	if !isUnstructured {
-		logger.V(3).Infof("Deployment watch received unknown object type %q",
-			reflect.TypeOf(deployment))
+		logger.V(3).Infof("Deployment watch received unknown object type %T",
+			event.Object)
 		return
 	}
 
 	// Start over, prove that rollout is complete.
 	dia.deploymentErrors = map[string]string{}
 
 	// Do nothing if this is not the Deployment we're waiting for.
-	if deployment.GetName() != inputDeploymentName {
+	if deployment.GetName() != inputDeploymentName || deployment.GetNamespace() != dia.config.currentInputs.GetNamespace() {
```

Comment:
Since the informer is already filtering by namespace, is this necessary?

Reply:
Yup - makes sense. Will remove.